### PR TITLE
GUI: Fix recursion of booting inside GetBootConfirmation()

### LINKED ISSUE #13800: Spamming leftclick on gamelist crashes RPCS3
Spamming left doubleclick on the game list crashes RPCS3 when the dialogs in GetBootConfirmation are configured as "do not ask again".

### PR DESCRIPTION
**Pull request by elad335**
* Deprecate more QCoreApplication::processEvents() usages. Followup to #13876.
* Fix recursion of booting inside GetBootConfirmation(). Fixes #13800 
* Fix emulator args restoration on failed Emulator::BootGame(), queue the restoration to Emulator::Kill() if emulation sop is pending.

### CODE REVIEW

File: rpcs3/rpcs3qt/main_window.cpp [hunk 1]
```diff
@@ -1629,18 +1726,27 @@ void main_window::HandlePupInstallation(const QString& file_path, const QString&
 		});
 
 		// Wait for the completion
-		for (uint value = progress.load(); value < update_filenames.size(); std::this_thread::sleep_for(5ms), value = progress)
+		uint value{};
```
Comment (Megamouse) [inline rpcs3/rpcs3qt/main_window.cpp:1729]: Why is this not a local variable in the lambda?

File: rpcs3/rpcs3qt/gui_settings.cpp [hunk 2]
```diff
@@ -193,11 +195,22 @@ void gui_settings::ShowInfoBox(const QString& title, const QString& text, const
 
 bool gui_settings::GetBootConfirmation(QWidget* parent, const gui_save& gui_save_entry)
 {
-	while (Emu.GetStatus(false) == system_state::stopping)
+	// Ensure no game has booted inbetween
+	const auto guard = Emu.MakeEmulationStateGuard();
+
+	auto info = Emu.GetEmulationIdentifier();
```
Comment (Megamouse) [inline rpcs3/rpcs3qt/gui_settings.cpp:201]: const

File: rpcs3/rpcs3qt/gui_settings.cpp [hunk 3]
```diff
@@ -193,11 +195,22 @@ void gui_settings::ShowInfoBox(const QString& title, const QString& text, const
...
+	auto old_status = Emu.GetStatus(false);
```
Comment (Megamouse) [inline rpcs3/rpcs3qt/gui_settings.cpp:202]: const

File: rpcs3/rpcs3qt/main_window.cpp [hunk 4]
```diff
@@ -1369,9 +1463,12 @@ void main_window::ExtractTar()
 			error += file;
 		}
 
-		pdlg.SetValue(pdlg.value() + 1);
-		QApplication::processEvents();
-	}
+		pdlg_progress += 1;
```
Comment (Megamouse) [inline rpcs3/rpcs3qt/main_window.cpp:1466]:
```suggestion
		pdlg_progress++;
```

File: rpcs3/rpcs3qt/main_window.cpp [hunk 5]
```diff
@@ -1066,9 +1148,13 @@ bool main_window::HandlePackageInstallation(QStringList file_paths, bool from_bo
 	pdlg.show();
 
 	// Wait for the completion
-	for (usz i = 0, set_text = umax; i < readers.size() && result.error == package_install_result::error_type::no_error;)
+	process_qt_events(5, [&, i = 0, set_text = -1, readers_size = ::narrow<int>(readers.size())]() mutable
```
Comment (Megamouse) [inline rpcs3/rpcs3qt/main_window.cpp:1151]: This breaks set_text and i

File: rpcs3/rpcs3qt/main_window.cpp [hunk 6]
```diff
@@ -525,12 +563,56 @@ void main_window::show_boot_error(game_boot_result status)
...
+	if (status != game_boot_result::still_running)
+	{
+		msg->setAttribute(Qt::WA_DeleteOnClose);
+		return;
+	}
+
+	// Automatically close for still_running error
+	std::shared_ptr<std::function<void()>> check_func;
+
+	auto info = Emu.GetEmulationIdentifier();
```
Comment (Megamouse) [inline rpcs3/rpcs3qt/main_window.cpp:578]: Const

File: rpcs3/rpcs3qt/main_window.cpp [hunk 7]
```diff
@@ -525,12 +563,56 @@ void main_window::show_boot_error(game_boot_result status)
...
+	msg->setParent(this);
 	msg->open();
+
+	if (status != game_boot_result::still_running)
+	{
+		msg->setAttribute(Qt::WA_DeleteOnClose);
+		return;
+	}
+
+	// Automatically close for still_running error
```
Comment (Megamouse) [inline rpcs3/rpcs3qt/main_window.cpp:575]: What does this complicated spaghetti code achieve and why is it necessary?

File: rpcs3/rpcs3qt/main_window.cpp [hunk 8]
```diff
@@ -96,15 +99,49 @@ std::shared_ptr<CPUDisAsm> make_basic_ppu_disasm();
...
 		// I noticed that calling this from an Emu callback can cause the
 		// caller to get stuck for a while during newly opened Qt dialogs.
 		// Adding a timeout here doesn't seem to do anything in that case.
-		QApplication::processEvents();
+		QEventLoop event_loop;
+
+		std::shared_ptr<std::function<void()>> check_iteration;
+		check_iteration = std::make_shared<std::function<void()>>([&]()
+		{
+			if (exit_cond())
```
Comment (Megamouse) [inline rpcs3/rpcs3qt/main_window.cpp:115]: Zero sanity check for exit cond

File: rpcs3/rpcs3qt/main_window.cpp [hunk 9]
```diff
@@ -96,15 +99,49 @@ std::shared_ptr<CPUDisAsm> make_basic_ppu_disasm();
 
 inline std::string sstr(const QString& _in) { return _in.toStdString(); }
 
-extern void process_qt_events()
+extern void process_qt_events(int repeat_duration, std::function<bool()> exit_cond)
```
Comment (Megamouse) [inline rpcs3/rpcs3qt/main_window.cpp:102]: Missing time suffix for duration, same in all extern declarations

File: rpcs3/rpcs3qt/main_window.cpp (hunk 1 above)
Comment (elad335) [reply rpcs3/rpcs3qt/main_window.cpp]: Fixed

File: rpcs3/rpcs3qt/main_window.cpp (hunk 7 above)
Comment (elad335) [reply rpcs3/rpcs3qt/main_window.cpp]: Removed

File: rpcs3/rpcs3qt/main_window.cpp (hunk 8 above)
Comment (elad335) [reply rpcs3/rpcs3qt/main_window.cpp:117]: Added

File: rpcs3/Emu/System.h [hunk 10]
```diff
@@ -206,8 +209,13 @@ class Emulator final
 	enum class stop_counter_t : u64{};
 
 	// Returns a different value each time we start a new emulation.
-	stop_counter_t GetEmulationIdentifier() const
+	stop_counter_t GetEmulationIdentifier(bool substract_one = false) const
```
Comment (Megamouse) [inline rpcs3/Emu/System.h:212]:
```suggestion
	stop_counter_t GetEmulationIdentifier(bool subtract_one = false) const
```

**Pull request merged**